ui-v2/app/utils/get-object-pool.js

```diff
@@ -0,0 +1,52 @@
+export default function(dispose = function() {}, max, objects = []) {
+  return {
+    acquire: function(obj, id) {
```

Comment:
Maybe this is just me being pedantic, but calling this an object pool when objects are provided directly is a misnomer. The core value of a traditional object pool is to prevent allocating and destroying tons of objects. Since this data structure isn't doing any form of object recycling, it isn't really an object pool. 

This is still a valuable structure by all means, I'm just not sure what to name it. `DisposableSet`? `DestructorCollection`? That one sounds Java as h*ck.

Reply:
Hmm yeah I'm not 100% sure on the vocabulary here either. The main reason it's not a 'true' pool is that we don't really have the access via `ember-data` API's currently to be able to do that. Ideally I'd like to turn it into a 'true' pool though, it all depends on a possible future `HTTPAdapter` and how that might work.

You're right though, the actual functionality here is about disposing/destroying objects so a lot of the vocab works, I think its mainly the filename. Let me rethink it a little, right now I'm tempted to stick to the 'pool' idea as ideally thats what I'd like it to be along with `HTTPAdapter`, but I'll give it another thought when I go back over.